crmd/notify.c

```diff
@@ -20,154 +20,721 @@
+
+        if (notify_script) {
+            entry = (notify_entry_t) {
+                .id = (char *) "legacy_notification",
+                .path = notify_script,
+                .timeout = CRMD_NOTIFY_DEFAULT_TIMEOUT_MS,
+                .recipient = notify_target
+            };
+            add_dup_notify_list_entry(&entry);
+            crm_notice("Legacy Notifications enabled");
+        }
+
+        return;
+    } else {
+        crm_notice("We have an alerts section in the cib");
+
+        if (notify_script) {
+            crm_warn("Cib contains configuration for Legacy Notifications "
+                     "which is overruled by alerts section");
+        }
+    }
+
+    if ((notify) && 
+        (crm_element_name(notify)) && 
+        (strcmp(crm_element_name(notify), XML_CIB_TAG_NOTIFY) != 0)) {
+        notify = first_named_child(notify, XML_CIB_TAG_NOTIFY);
+    }
+
+    for (; notify; notify = __xml_next(notify)) {
+        xmlNode *recipient;
+        int recipients = 0, envvars = 0;
+
+        entry = (notify_entry_t) {
+            .id = (char *) crm_element_value(notify, XML_NOTIFY_ATTR_ID),
+            .path = (char *) crm_element_value(notify, XML_NOTIFY_ATTR_PATH),
+            .timeout = crm_get_msec(crm_element_value(notify, 
+                                    XML_NOTIFY_ATTR_TIMEOUT)),
+            .tstamp_format = (char *) crm_element_value(notify, 
+                             XML_NOTIFY_ATTR_TSTAMP_FORMAT)
+        };
+        if (entry.timeout <= 0) {
+            entry.timeout = CRMD_NOTIFY_DEFAULT_TIMEOUT_MS;
+        }
+
+        entry.environment_variables =
+            get_envvars_from_cib(notify, 
+                                 entry.environment_variables,
+                                 &envvars);
+
+        crm_notice("Found notification: id=%s, path=%s, timeout=%d, "
+                   "tstamp_format=%s, %d additional environment variables", 
+                   entry.id, entry.path, entry.timeout,
+                   entry.tstamp_format, envvars);
+
+        for (recipient = first_named_child(notify, 
+                                           XML_CIB_TAG_NOTIFY_RECIPIENT); 
+             recipient; recipient = __xml_next(recipient)) {
+            int envvars_added = 0;
+
+            entry.recipient = (char *) crm_element_value(recipient, 
+                                                XML_NOTIFY_ATTR_REC_VALUE);
+            recipients++;
+
+            entry.environment_variables =
+                get_envvars_from_cib(recipient, 
+                                     entry.environment_variables, 
+                                     &envvars_added);
+
+            add_dup_notify_list_entry(&entry);
+
+            crm_notice("Notification has recipient: id=%s, value=%s, "
+                       "%d additional environment variables", 
+                       crm_element_value(recipient, XML_NOTIFY_ATTR_REC_ID), 
+                       entry.recipient, envvars_added);
```

Comment:
pcs: when adding a recipient to an alert then all the attributes that were allowed in the instance_attributes and the meta_attributes section are allowed again.
Implementation is easy because it just calls the code for the alert a 2nd time.
Current state is that just the overruling attributes are in the instance_attributes and meta_attributes section of the recipient section. So it can easily be switched on and
off by just changing the schema.